NetworkLayer/Model/Ad.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  Ad.swift
+//  NetworkLayer
+//
+//  Created by Pierre on 15/11/2022.
+//
+
+import Foundation
+
+struct Ad: Codable {
```

Comment:
```suggestion
struct Ad: Decodable {
```

Utilise "Decodable" tu cherches simplement à décoder.